homeassistant/components/devolo_home_network/image.py

```diff
@@ -0,0 +1,102 @@
+"""Platform for image integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from functools import partial
+from typing import Any
+
+from devolo_plc_api import Device, wifi_qr_code
+from devolo_plc_api.device_api import WifiGuestAccessGet
+
+from homeassistant.components.image import ImageEntity, ImageEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN, IMAGE_GUEST_WIFI, SWITCH_GUEST_WIFI
+from .entity import DevoloCoordinatorEntity
+
+QR_CODE_SCALE = 4
+
+
+@dataclass
+class DevoloImageRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    image_func: Callable[[WifiGuestAccessGet], bytes]
+
+
+@dataclass
+class DevoloImageEntityDescription(
+    ImageEntityDescription, DevoloImageRequiredKeysMixin
+):
+    """Describes devolo image entity."""
+
+
+IMAGE_TYPES: dict[str, DevoloImageEntityDescription] = {
+    IMAGE_GUEST_WIFI: DevoloImageEntityDescription(
+        key=IMAGE_GUEST_WIFI,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        image_func=partial(wifi_qr_code, scale=QR_CODE_SCALE),
```

Comment:
Do you know if the scale vector is required?

As far as I understood, the generated QR code is an SVG and we don't need any scale there.

Reply:
I actually chose SVG exactly for that reason, but without the scale it's kind of tiny. 

![grafik](https://github.com/home-assistant/core/assets/20171515/7403be7e-11de-4a1b-b535-c0f2a8bf3593)

Reply:
I spoke quickly with the frontend team.

Your SVG has a `width` and `height` set and therefore the frontend respects those values.
We suggest you use a `viewbox` (e.g. `viewBox="0 0 37 37"`) instead of those attributes.
![image](https://github.com/home-assistant/core/assets/26537646/b29f4ace-e525-4ab6-b92b-92a739cb858f)


Can you please try this?

Reply:
That did the trick. :+1: